wp/lib/bap_wp/src/constraint.ml

```diff
@@ -49,6 +50,13 @@ let get_model_exn (solver : Solver.solver) : Model.model =
 let goal_to_string (g : goal) : string =
   Format.sprintf "%s: %s%!" g.goal_name (Expr.to_string (Expr.simplify g.goal_val None))
 
+let expr_to_hex (exp : z3_expr) : string =
+  let decimal = exp |> BV.numeral_to_string |> Int.of_string in
+  let size = exp |> Expr.get_sort |> BV.get_size |> Int.to_float in
+  (* Calculate how many 0s to pad the output with. *)
```

Comment:
I guess this works for small numbers, but I don't understand why you don't use integer division and remainder... Like say `if size mod 4 = 0 then size/4 else size/4 + 1`. IDK

Floats make me nervous. Not really crucial though, since we're using this for printing.

Reply:
That honestly didn't come to mind when I wrote it. Having floats does make me nervous as well.

Reply:
I just wrote that off the cuff though, you might want to test it...

Reply:
Yep, tested it and it works!

Reply:
👍